smolyay/samples.py

```diff
@@ -0,0 +1,409 @@
+            )
+        else:
+            points = numpy.zeros(1)
+        self._points = self.scale_to_domain(points, [-1, 1], self.domain)
+
+
+class NestedClenshawCurtisPointSet(NestedUnidimensionalPointSet):
+    r"""Generate nested Clenshaw Curtis points
+
+    The :attr:`points` for this interpolation scheme are the extrema of the
+    Chebyshev polynomials of the first kind on the domain :math:`[-1, 1]`:
+
+    .. math::
+
+        x_i^* = -\cos(\pi i/n), i = 0,...,n
+
+    For the special case :math:`n = 0`, there is only one point :math:`x_0^* = 0`.
+
+    The nested Clenshaw Curtis points come from the nested extrema of the
+    Chebyshev polynomials of the first kind :math:`n` where n is part of a 
+    sequence
+     
+    .. math::
+
+    n = \begin{cases}
+         0 & \text{ if } L = 0\\ 
+         2^{L} & \text{ if } L > 0 
+    \end{cases}
+
+    where L is a whole number.
+
+    As the extrema of the Chebyshev polynomials of degree :math:`n` are nested,
+    meaning any :math:`n` will have extrema at the same points as every n that
+    precedes it in the sequence. The extrema are organized into levels such that
+    a level L will contain the extrema of Chebyshev polynomial n(L) that are
+    not the extrema of any Chebyshev polynomial n(k) where k is a whole number
+    and k < L.
+
+    Parameters
+    ----------
+    max_level ; int
+        the maximum level the points are used for
+    """
+
+    def __init__(self, domain, max_level):
+        super().__init__(domain, max_level)
+
+    def _create(self):
+        r"""Create the points in the set.
+
+        Generating nested extrema of chebyshev polynomials of the first kind.
+        """
+        # create properties for levels
+        rule = lambda x: 1 if x == 0 else 2**x + 1
+        self._num_points_per_level = numpy.ones(self.max_level + 1, dtype=int)
+        self._num_points_per_level[1:] = [
+            rule(i) - rule((i - 1)) for i in range(1, self.max_level + 1)
+        ]
+
+        self._start_level = numpy.zeros_like(self._num_points_per_level)
+        self._start_level[1:] = numpy.cumsum(self._num_points_per_level[:-1])
+
+        self._end_level = self._start_level + self._num_points_per_level
+        # points
+        points = numpy.zeros(numpy.sum(self._num_points_per_level))
+        degree = 0
+        num_levels = self.max_level + 1
+        for i in range(1, num_levels):
+            degree = 2**i
+            # find indexes of extrema not already found. Fraction index/degree
+            # cannot be further simplified
+            if i == 1:
+                indexes = numpy.arange(0, degree + 1, 2, dtype=int)
+            else:
+                indexes = indexes = numpy.arange(1, degree, 2, dtype=int)
+                indexes = indexes[~(numpy.gcd(indexes, degree) > 1)]
+            # generate extrema
+            new_points = list(-numpy.cos(numpy.pi * indexes / degree))
+            points[self._start_level[i] : self._end_level[i]] = new_points
```

Comment:
No need to go through a list
```suggestion
            points[self._start_level[i] : self._end_level[i]] = -numpy.cos(numpy.pi * indexes / degree)
```